torch/distributed/distributed_c10d.py

```diff
@@ -191,16 +183,25 @@ def _store_based_barrier(rank, store, timeout):
     """
     store_key = "{}:{}".format(STORE_BASED_BARRIER_PREFIX, _group_count)
     store.add(store_key, 1)
+    logging.info('Added key: {} to store for rank: {}'.format(store_key, rank))
 
     # Now wait for all workers to check in with the store.
     world_size = get_world_size()
-    worker_count = int(store.get(store_key))
+    # Use 'add' instead of 'get' since for some store implementations 'add'
+    # doesn't work well with 'get'. Ideally the store implementations should
+    # be fixed, but for backward compatiblity reasons it is risky to change
+    # the store implementations. Once, we completely migrate away from these
+    # legacy stores, we can use 'get' here instead.
```

Comment:
Curious - which store implementations display this behavior, and what happens when 'get' is called? Since the store docs were only added in 1.7, I'm guessing not that many OSS users use the store directly so we should be alright fixing the store implementations wrt backward compatibility right?

Reply:
Never mind, just saw your comment regarding the Caffe2 stores